develop/interoperability/xcm-runtime-apis.md

```diff
@@ -0,0 +1,360 @@
+
+    ```js
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/call-example.js'
+    ```
+
+    !!!note
+        Make sure to replace `INSERT_USER_ADDRESS` with your SS58 address before running the script.
+
+    ***Output***
+
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/call-example-output.html::11'
+                ...
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/call-example-output.html:189'
+
+    ---
+
+### Dry Run XCM
+
+This API allows to directly dry-run an xcm message instead of an extrinsic and check if it will execute succesfully and what other xcm messages will be forwarded to other chains.
+
+```rust
+--8<-- 'https://raw.githubusercontent.com/paritytech/polkadot-sdk/refs/heads/stable2412/polkadot/xcm/xcm-runtime-apis/src/dry_run.rs:70:70'
+```
+
+??? interface "Input parameters"
+
+    `origin_location` ++"VersionedLocation"++ <span class="required" markdown>++"required"++</span>
+
+    The location of the origin that will execute the xcm message.
+
+    ---
+
+    `xcm` ++"VersionedXcm<Call>"++ <span class="required" markdown>++"required"++</span>
+
+    A versioned XCM message.
+
+    ---
+
+??? interface "Output parameters"
+
+    ++"Result<XcmDryRunEffects<Event>, Error>"++
+
+    Effects of dry-running an extrinsic. If an error occurs, it is returned instead of the effects.
+
+    ??? child "Type `CallDryRunEffects<Event>`"
+
+        `execution_result` ++"DispatchResultWithPostInfo"++
+
+        The result of executing the extrinsic.
+
+        ---
+
+        `emitted_events` ++"Vec<Event>"++
+
+        The list of events fired by the extrinsic.
+
+        ---
+
+        `forwarded_xcms` ++"Vec<(VersionedLocation, Vec<VersionedXcm<()>>)>"++
+
+        The list of XCMs that were queued for sending.
+
+    ---
+
+??? interface "Example"
+
+    This example demonstrates how to simulate a teleport asset transfer from the Paseo network to the Paseo Asset Hub parachain. The code shows how to test and verify the received XCM message's behavior in the destination chain through a dry run on the live network.
+
+     ***Usage with PAPI***
+
+    ```js
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/xcm-example.js'
+    ```
+
+    !!!note
+        Make sure to replace `INSERT_USER_ADDRESS` with your SS58 address before running the script.
+
+    ***Output***
+
+    --8<-- 'code/develop/interoperability/xcm-runtime-apis/xcm-example-output.html'
+
+    ---
+
+## XCM Payment API
+
+The [XCM Payment API](https://paritytech.github.io/polkadot-sdk/master/xcm_runtime_apis/fees/index.html){target=\_blank} provides a standardized way to determine the costs and payment options for executing XCM messages. Specifically, it enables clients to:
+
+- Retrieve the weight required to execute an XCM message
+- Obtain a list of acceptable `AssetIds` for paying execution fees
+- Calculate the cost of the weight in a specified `AssetId`
+- Estimate the fees for XCM message delivery
```

Comment:
These are the individual functions exposed by the API, but what I'd put is just two things:
- Getting execution fees
- Getting delivery fees

The first use case makes use of those first 3 functions, but that's more lower level

Reply:
Also, the first time we mention "weight", can we link to the concept of weight? Maybe give it a little intro. It's not such a common concept that it's 2D and not just called "gas"

Reply:
The scope of this page is to document the existing APIs themselves. I'm working on a tutorial that will use them together, and this tutorial will be linked from this page.